runtime/altair/src/lib.rs

```diff
@@ -926,6 +928,76 @@ impl orml_tokens::Config for Runtime {
 	type DustRemovalWhitelist = frame_support::traits::Nothing;
 }
 
+impl orml_asset_registry::Config for Runtime {
+	type Event = Event;
+	type CustomMetadata = CustomMetadata;
+	type AssetId = CurrencyId;
+	type AuthorityOrigin = AssetAuthority;
+	type AssetProcessor = CustomAssetProcessor;
+	type Balance = Balance;
+	type WeightInfo = ();
+}
+
+pub struct AssetAuthority;
+impl EnsureOriginWithArg<Origin, Option<CurrencyId>> for AssetAuthority {
+	type Success = ();
+
+	fn try_origin(origin: Origin, asset_id: &Option<CurrencyId>) -> Result<Self::Success, Origin> {
+		match asset_id {
+			// Only the pools pallet should directly register/update tranche tokens
+			Some(CurrencyId::Tranche(_, _)) => Err(origin),
+
+			// Any other `asset_id` defaults to EnsureRoot
+			_ => EnsureRoot::try_origin(origin),
+		}
+	}
+
+	#[cfg(feature = "runtime-benchmarks")]
+	fn successful_origin(_: &Option<CurrencyId>) -> Origin {
+		todo!()
+	}
+}
+
+//TODO(nuno): move this to common-types
+#[derive(
+	Clone, Copy, PartialOrd, Ord, PartialEq, Eq, Debug, Encode, Decode, TypeInfo, MaxEncodedLen,
+)]
+pub struct CustomMetadata;
```

Comment:
`todo`:

- [ ] do we want a custom metadata per runtime, maybe? if so, leave this here.
- [ ] we need to agree on what values we want to add to it.